pkg/da/data_availability_header.go

```diff
@@ -38,10 +38,23 @@ type DataAvailabilityHeader struct {
 }
 
 // NewDataAvailabilityHeader generates a DataAvailability header using the provided square size and shares
-func NewDataAvailabilityHeader(squareSize uint64, shares [][]byte) (DataAvailabilityHeader, error) {
+func NewDataAvailabilityHeader(eds *rsmt2d.ExtendedDataSquare) DataAvailabilityHeader {
+	// generate the row and col roots using the EDS
+	dah := DataAvailabilityHeader{
+		RowsRoots:   eds.RowRoots(),
+		ColumnRoots: eds.ColRoots(),
+	}
+
+	// generate the hash of the data using the new roots
+	dah.Hash()
+
+	return dah
+}
+
+func ExtendShares(squareSize uint64, shares [][]byte) (*rsmt2d.ExtendedDataSquare, error) {
 	// Check that square size is with range
 	if squareSize < consts.MinSquareSize || squareSize > consts.MaxSquareSize {
```

Comment:
when extending the shares into an extended data square, check that the square size is within the min and the max, and that the shares provided are formatted as expected.

Reply:
the diff make it's difficult to see the differences... I reccomend simply looking at the file. The previous `NewDataAvailabilityHeader` function has been split in two. This provides the user with direct access to the extended data square.
https://github.com/celestiaorg/celestia-core/blob/422eb4d8f161aa1cd1a1b7752ac3c2b35c20c66c/pkg/da/data_availability_header.go#L40-L74